subseas_flex_fcst/maproom.py

```diff
@@ -0,0 +1,468 @@
+
+
+def get_coords(click_lat_lng):
+    if click_lat_lng is not None:
+        return click_lat_lng
+    else:
+        fcst_mu = cptio.open_cptdataset(Path(DATA_PATH, Path(CONFIG["forecast_mu_file"])))
+        return [(fcst_mu.Y[0].values+fcst_mu.Y[-1].values)/2, (fcst_mu.X[0].values+fcst_mu.X[-1].values)/2]
+
+
+@APP.callback(Output("map", "click_lat_lng"), Input("submitLatLng","n_clicks"), State("latInput", "value"), State("lngInput", "value"))
+def inputCoords(n_clicks,latitude,longitude):
+    if latitude is None:
+        return None
+    else:
+        lat_lng = [latitude, longitude]
+        return lat_lng
+
+
+@APP.callback(Output("layers_group", "children"), Input("map", "click_lat_lng"))
+def map_click(click_lat_lng):
+    lat_lng = get_coords(click_lat_lng)
+
+    return dlf.Marker(
+        position=lat_lng, children=dlf.Tooltip("({:.3f}, {:.3f})".format(*lat_lng))
+    )
+
+
+@APP.callback(
+    Output("percentile_style", "style"),
+    Output("threshold_style", "style"),
+    Input("variable", "value")
+)
+def display_relevant_control(variable):
+
+    displayed_style={
+        "position": "relative",
+        "width": "190px",
+        "display": "flex",
+        "padding": "10px",
+        "vertical-align": "top",
+    }
+    if variable == "Percentile":
+        style_percentile=displayed_style
+        style_threshold={"display": "none"}
+    else:
+        style_percentile={"display": "none"}
+        style_threshold=displayed_style
+    return style_percentile, style_threshold
+
+
+@APP.callback(
+    Output("cdf_graph", "figure"),
+    Output("pdf_graph", "figure"),
+    Input("map", "click_lat_lng"),
+)
+def local_plots(click_lat_lng):
+
+    lat, lng = get_coords(click_lat_lng)
+
+    # Reading
+
+    fcst_mu, fcst_var, obs, hcst = read_cptdataset(y_transform=CONFIG["y_transform"])
+
+    # Spatial Tolerance for lat/lon selection clicking on map
+    half_res = (fcst_mu["X"][1] - fcst_mu["X"][0]) / 2
+    tol = np.sqrt(2 * np.square(half_res)).values 
+    
+    # Errors handling
+    try:
+        isnan = (np.isnan(fcst_mu.sel(
+            X=lng, Y=lat, method="nearest", tolerance=tol
+        )).sum()) + (np.isnan(obs.sel(
+            X=lng, Y=lat, method="nearest", tolerance=tol
+        )).sum())
+        if CONFIG["y_transform"]:
+            isnan_yt = (np.isnan(hcst.sel(
+                X=lng, Y=lat, method="nearest", tolerance=tol
+            )).sum())
+            isnan = isnan + isnan_yt
+        if isnan > 0:
+            errorFig = pingrid.error_fig(error_msg="Data missing at this location")
+            return errorFig, errorFig
+    except KeyError:
+        errorFig = pingrid.error_fig(error_msg="Grid box out of data domain")
+        return errorFig, errorFig
+    
+    fcst_mu = fcst_mu.sel(X=lng, Y=lat, method="nearest", tolerance=tol)
+    fcst_var = fcst_var.sel(X=lng, Y=lat, method="nearest", tolerance=tol)
+    obs = obs.sel(X=lng, Y=lat, method="nearest", tolerance=tol)
```

Comment:
These three `sel`s are repeats of ones that were done inside the try/except. There's a risk that we'll change one and forget to change the other. I would move these into the try/except, e.g.
```
try:
    fcst_mu = fcst_mu.sel(X=lng, Y=lat, method="nearest", tolerance=tol)
    isnan = (np.isnan(fcst_mu).sum())
    ...
```